packages/node/src/streams/lengthChunkSplitter.ts

```diff
@@ -0,0 +1,51 @@
+import { ChunkSplitter } from './chunkifier.js';
+
+/**
+ * Splits data into chunks with maximum length.
+ * @param maxLength Maximum length of one chunk.
+ * @param wholeLines If `true`, will split chunks before newlines, so whole lines are passed to the chunk.
+ */
+export function lengthChunkSplitter(maxLength: number, wholeLines: 'skip' | 'break' | false = false): ChunkSplitter {
+    let seen = 0;
+
+    const emptyBuffer = Buffer.of();
+
+    return function lengthChunkSplitter(data) {
+        const remainingLength = maxLength - seen;
+        if (data.length <= remainingLength) {
```

Comment:
its weird that later we respect wholeLines information but when the data.length <= remainingLenght we don't. Actually why?

Reply:
because if all data fits into `remainingLength`, we don't need to split it. Ergo, line splitting does not occur.